core/client/fs/src/main/java/alluxio/client/file/MetadataCachingBaseFileSystem.java

```diff
@@ -73,12 +79,78 @@ public MetadataCachingBaseFileSystem(FileSystemContext context) {
         TimeUnit.SECONDS, new SynchronousQueue<>());
   }
 
+  @Override
+  public void createDirectory(AlluxioURI path, CreateDirectoryPOptions options)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, AlluxioException {
+    super.createDirectory(path, options);
+    mMetadataCache.invalidate(path.getParent());
+    mMetadataCache.invalidate(path);
```

Comment:
shall we invalidate the cache first before modifying file system, to be conservative?
```suggestion
    mMetadataCache.invalidate(path.getParent());
    mMetadataCache.invalidate(path);
    super.createDirectory(path, options);

```